### PR TITLE
Implement Command record to manage commands

### PR DESCRIPTION
**Pull request by AnidemDex**
This PR aims to add CommandRecord class, a "singleton" that will take care about the commands registered in plugin, so other objects can make usage of this singleton to register their own commands or know what commands are registered.

## How?
Get `CommandRecord` with `Blockflow.CommandRecord.get_record()`, always verify that is a valid object.

Use `register` to add a command to the list, and `unregister` to do the opposite. Use `reload_from_project_settings` to force the record to build itself again using the `ProjectSettings` registered values.

If you want to know that record changed, listen to `command_list_changed` signal.

## Why?
I'm not the only one that is going to be adding commands to the plugin, and I don't want to introduce _hard-linked_ stuff to _others_ code, so I decided to just group and expose the part that handles the command registry in editor.

I'm gonna use it to create command templates and update the command list, and didn't wanted the `CommandList` node handle all this stuff.

### CODE REVIEW

**Comment by AnidemDex** [comment]
Don't merge yet, I'm gonna remove the "as template" argument, is no needed

**Comment by AnidemDex** [comment]
Ok, all done. If you can review this before merging I would appreciate it @Crystalwarrior :pray:

**Review by Crystalwarrior — APPROVED** [review]

**Pull request merged**
